src/blockchains/construct_worker.ts

```diff
@@ -0,0 +1,34 @@
+'use strict';
+import { BaseWorker } from '../lib/worker_base';
+import { CardanoWorker } from './cardano/cardano_worker';
+import { ERC20Worker } from './erc20/erc20_worker';
+import { ETHWorker } from './eth/eth_worker';
+import { ETHBlocksWorker } from './eth_blocks/eth_blocks_worker';
+import { MaticWorker } from './matic/matic_worker';
+import { ReceiptsWorker } from './receipts/receipts_worker';
+import { UTXOWorker } from './utxo/utxo_worker';
+import { XRPWorker } from './xrp/xrp_worker';
+
+export function constructWorker(blockchain: string, settings: any): BaseWorker {
+  switch (blockchain) {
+    case 'cardano':
+      return new CardanoWorker(settings);
+    case 'erc20':
+      return new ERC20Worker(settings);
+    case 'eth':
+      return new ETHWorker(settings);
+    case 'eth_blocks':
+      return new ETHBlocksWorker(settings);
+    case 'matic':
+      return new MaticWorker(settings);
+    case 'receipts':
+      return new ReceiptsWorker(settings);
+    case 'utxo':
+      return new UTXOWorker(settings);
+    case 'xrp':
+      return new XRPWorker(settings);
+    default:
+      throw Error(`Blockchain type '${blockchain}' is not recognized`);
```

Comment:
there is no need for error if blockchain is enum or union type which is preffered in this lang

Reply:
But I still need to construct the enum out of the string ENV variable in a similar manner right?